sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -219,7 +221,7 @@ object TableOutputResolver extends SQLConfHelper with Logging {
       fillDefaultValue: Boolean = false): Seq[NamedExpression] = {
     val matchedCols = mutable.HashSet.empty[String]
     val reordered = expectedCols.flatMap { expectedCol =>
-      val matched = inputCols.filter(col => conf.resolver(col.name, expectedCol.name))
+      val matched = inputCols.filter(col => resolver(col.name, expectedCol.name))
```

Comment:
This `conf` is given by the method parameter. I'm not sure this is correct optimization.